packages/headless/src/features/facets/category-facet-set/category-facet-set-selectors.ts

```diff
@@ -15,7 +16,7 @@ function isCategoryFacetResponse(
 }
 
 export const categoryFacetResponseSelector = (
-  state: CategoryFacetSection & SearchSection,
+  state: CategoryFacetSection & Partial<SearchSection | ProductListingSection>,
```

Comment:
Is this type change needed?

It seems like both `SearchSection` and `ProductListingSection` are optional?
I also don't see the change scaling nicely in the code-base as more use-cases are added.

We could store the union in a type to remove the duplication,

```
type CategoryFacetResponseSection = SearchSection | ProductListingSection;

state: CategoryFacetSection & CategoryFacetResponseSection
```

but if no change is needed, all the better,

Reply:
I've created a new type to store the union of search and product-listing section.
The type change is needed to make a distinction between a section coming from product-listing or search when coming from the analytics branch. This part can (or better must) be removed when the analytics have independent product-listing or search clients.